File: scripts/check-payment-trigger.js
#!/usr/bin/env node
/**
 * Check payment trigger status and payment #4226
 */

const { PrismaClient } = require('@prisma/client');

const prisma = new PrismaClient();

async function checkTrigger() {
  try {
    console.log('🔌 Connecting to database...\n');

    // Check if trigger function exists
    console.log('1️⃣ Checking trigger function...');
    const functionCheck = await prisma.$queryRaw`
      SELECT EXISTS(
        SELECT 1 FROM pg_proc 
        WHERE proname = 'generate_payment_id'
      ) as exists
    `;
    const functionExists = functionCheck[0]?.exists || false;
    console.log(functionExists ? '   ✅ Function exists' : '   ❌ Function NOT FOUND');

    // Check if trigger is attached
    console.log('\n2️⃣ Checking trigger on payments table...');
    const triggerCheck = await prisma.$queryRaw`
      SELECT tgname, tgenabled 
      FROM pg_trigger 
      WHERE tgrelid = 'payments'::regclass 
      AND tgname LIKE '%payment_id%'
    `;
    
    if (triggerCheck.length > 0) {
      console.log(`   ✅ Found ${triggerCheck.length} trigger(s):`);
      triggerCheck.forEach(t => {
        const status = t.tgenabled === 'O' ? 'ENABLED' : 'DISABLED';
        console.log(`      - ${t.tgname} (${status})`);
      });
    } else {
      console.log('   ❌ No trigger found');
    }

    // Check payment #4226
    console.log('\n3️⃣ Checking payment #4226...');
    const payment = await prisma.payment.findUnique({
      where: { id: 4226 },
      select: { id: true, paymentId: true, recordUuid: true }
    });

    if (payment) {
      console.log(`   Payment #${payment.id}:`);
      console.log(`      payment_id: ${payment.paymentId || '❌ NULL/EMPTY'}`);
      console.log(`      record_uuid: ${payment.recordUuid || '❌ NULL/EMPTY'}`);
      
      if (!payment.paymentId || !payment.recordUuid) {
        console.log('\n   ⚠️ Missing IDs - trigger was not active when created');
      }
    } else {
      console.log('   ❌ Payment #4226 not found');
    }

    // Check last 5 payments
    console.log('\n4️⃣ Last 5 payments:');
    const recentPayments = await prisma.payment.findMany({
      take: 5,
      orderBy: { id: 'desc' },
      select: { id: true, paymentId: true, recordUuid: true }
    });

    recentPayments.forEach(p => {
      const pidStatus = p.paymentId ? '✅' : '❌';
      const uuidStatus = p.recordUuid ? '✅' : '❌';
      console.log(`   ID ${p.id}: payment_id=${p.paymentId || 'NULL'} ${pidStatus}, uuid=${p.recordUuid || 'NULL'} ${uuidStatus}`);
    });

    console.log('\n' + '='.repeat(60));
    if (!functionExists || triggerCheck.length === 0) {
      console.log('❌ TRIGGER NOT INSTALLED');
      console.log('   Run: pwsh scripts/apply-payment-trigger.ps1');
    } else {
      console.log('✅ TRIGGER IS ACTIVE');
      console.log('   New payments will auto-generate IDs');
    }

  } catch (error) {
    console.error('\n❌ Error:', error.message);
    process.exit(1);
  } finally {
    await prisma.$disconnect();
  }
}

checkTrigger();

File: scripts/check-payment-trigger.py
#!/usr/bin/env python3
"""Check if payment trigger is properly installed and working"""

import os
import sys
import psycopg2

# Get DATABASE_URL from environment (should already be set)
DATABASE_URL = os.getenv('DATABASE_URL')
if not DATABASE_URL:
    print("❌ DATABASE_URL not found in environment")
    print("   Please set it first: $env:DATABASE_URL='your_connection_string'")
    sys.exit(1)

try:
    print("🔌 Connecting to database...")
    conn = psycopg2.connect(DATABASE_URL)
    cur = conn.cursor()
    
    # Check if trigger function exists
    print("\n1️⃣ Checking trigger function 'generate_payment_id'...")
    cur.execute("""
        SELECT EXISTS(
            SELECT 1 FROM pg_proc 
            WHERE proname = 'generate_payment_id'
        )
    """)
    function_exists = cur.fetchone()[0]
    
    if function_exists:
        print("   ✅ Function 'generate_payment_id' exists")
    else:
        print("   ❌ Function 'generate_payment_id' NOT FOUND")
    
    # Check if trigger is attached to payments table
    print("\n2️⃣ Checking trigger on 'payments' table...")
    cur.execute("""
        SELECT tgname, tgtype, tgenabled 
        FROM pg_trigger 
        WHERE tgrelid = 'payments'::regclass 
        AND tgname LIKE '%payment_id%'
    """)
    triggers = cur.fetchall()
    
    if triggers:
        print(f"   ✅ Found {len(triggers)} trigger(s):")
        for tgname, tgtype, tgenabled in triggers:
            status = "ENABLED" if tgenabled == 'O' else "DISABLED"
            print(f"      - {tgname} ({status})")
    else:
        print("   ❌ No payment_id trigger found on payments table")
    
    # Test if auto-generation works by checking recent payments
    print("\n3️⃣ Checking recent payments for auto-generated IDs...")
    cur.execute("""
        SELECT id, payment_id, record_uuid, created_at 
        FROM payments 
        ORDER BY id DESC 
        LIMIT 5
    """)
    recent_payments = cur.fetchall()
    
    if recent_payments:
        print(f"   📊 Last 5 payments:")
        for pid, payment_id, record_uuid, created_at in recent_payments:
            payment_id_status = "✅" if payment_id else "❌ EMPTY"
            record_uuid_status = "✅" if record_uuid else "❌ EMPTY"
            print(f"      ID {pid}: payment_id={payment_id or 'NULL'} {payment_id_status}, record_uuid={record_uuid or 'NULL'} {record_uuid_status}")
    else:
        print("   ℹ️ No payments found in table")
    
    # Check for payment #4226 specifically
    print("\n4️⃣ Checking payment #4226...")
    cur.execute("""
        SELECT id, payment_id, record_uuid 
        FROM payments 
        WHERE id = 4226
    """)
    payment_4226 = cur.fetchone()
    
    if payment_4226:
        pid, payment_id, record_uuid = payment_4226
        print(f"   Payment #{pid}:")
        print(f"      payment_id: {payment_id or '❌ NULL/EMPTY'}")
        print(f"      record_uuid: {record_uuid or '❌ NULL/EMPTY'}")
        
        if not payment_id or not record_uuid:
            print("\n   ⚠️ Payment #4226 has missing IDs - trigger was not active when it was created")
    else:
        print("   ❌ Payment #4226 not found")
    
    cur.close()
    conn.close()
    
    print("\n" + "="*60)
    if not function_exists or not triggers:
        print("❌ TRIGGER NOT PROPERLY INSTALLED")
        print("   Run: pwsh scripts/apply-payment-trigger.ps1")
    else:
        print("✅ TRIGGER IS INSTALLED AND ACTIVE")
        print("   New payments will auto-generate payment_id and record_uuid")
    
except Exception as e:
    print(f"\n❌ Error: {e}")
    sys.exit(1)
